addon/components/ember-table.js

```diff
@@ -0,0 +1,564 @@
+    var numFixedColumns = this.get('fixedColumns.length');
+    var columns = this.get('columns');
+    columns.removeObject(column);
+    columns.insertAt(numFixedColumns + newIndex, column);
+    return this.prepareTableColumns();
+  },
+
+  // An array of Ember.Table.Row computed based on `content`
+  bodyContent: Ember.computed(function() {
+    return RowArrayController.create({
+      target: this,
+      parentController: this,
+      container: this.get('container'),
+      itemController: Row,
+      content: this.get('content')
+    });
+  }).property('content.[]'),
+
+  // An array of Ember.Table.Row
+  footerContent: Ember.computed(function(key, value) {
+    if (value) {
+      return value;
+    } else {
+      return Ember.A();
+    }
+  }).property(),
+
+  fixedColumns: Ember.computed(function() {
+    var columns = this.get('columns');
+    if (!columns) {
+      return Ember.A();
+    }
+    var numFixedColumns = this.get('numFixedColumns') || 0;
+    return columns.slice(0, numFixedColumns) || [];
+  }).property('columns.@each', 'numFixedColumns'),
+
+  tableColumns: Ember.computed(function() {
+    var columns = this.get('columns');
+    if (!columns) {
+      return Ember.A();
+    }
+    var numFixedColumns = this.get('numFixedColumns') || 0;
+    return columns.slice(numFixedColumns, columns.get('length')) || [];
+  }).property('columns.@each', 'numFixedColumns'),
+
+  prepareTableColumns: function() {
+    var _this = this;
+    var columns = this.get('columns') || Ember.A();
+    columns.setEach('controller', this);
+    columns.forEach(function(col, i) {
+      col.set('nextResizableColumn', _this.getNextResizableColumn(columns, i));
+    });
+  },
+
+  getNextResizableColumn: function(columns, index) {
+    var column;
+    while (index < columns.length) {
+      index += 1;
+      column = columns.objectAt(index);
+      if (column != null && column.get('isResizable')) {
+        return column;
+      }
+    }
+    return null;
+  },
+
+  // ---------------------------------------------------------------------------
+  // View concerns
+  // ---------------------------------------------------------------------------
+
+  didInsertElement: function() {
+    this._super();
+    this.set('_tableScrollTop', 0);
+    this.elementSizeDidChange();
+    this.doForceFillColumns();
+  },
+
+  onResizeEnd: function() {
+    // We need to put this on the run loop, because resize event came from
+    // window. Otherwise, we get this warning when used in tests. You have
+    // turned on testing mode, which disabled the run-loop's autorun. You
+    // will need to wrap any code with asynchronous side-effects in an
+    // Ember.run
+    if (this.tableWidthNowTooSmall()) {
+      this.set('columnsFillTable', true);
+    }
+    return Ember.run(this, this.elementSizeDidChange);
+  },
+
+  elementSizeDidChange: function() {
+    if ((this.get('_state') || this.get('state')) !== 'inDOM') {
+      return;
```

Comment:
Seems valid here, since it short circuits the rest of the logic. Might be better to invert the logic and wrap the rest of the function it though.